src/components/IconButton.vue

```diff
@@ -0,0 +1,130 @@
+<script lang="ts">
+import { defineComponent, Prop, ref } from "vue";
+
+type Props = {
+  onClick: Function;
+  size: string;
+  color: string;
+  iconName: string;
+  pauseActiveStyle: boolean;
+};
+
+export default defineComponent({
+  name: "IconButton",
+  props: {
+    onClick: {
+      type: Function,
+      required: true,
+    },
+    size: {
+      type: String,
+      default: "large",
+      validator: function (value: string) {
+        return ["small", "medium", "large"].includes(value);
+      },
+    },
+    color: {
+      type: String,
+      default: "normal",
+      varidator: function (value: string) {
+        return ["normal", "danger"].includes(value);
+      },
+    },
+    iconName: {
+      type: String,
+      default: "",
+    },
+    pauseActiveStyle: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  setup: (props: Props, { emit }) => {
+    const isActive = ref(false);
+
+    const handleClick = (e: any) => {
```

Comment:
せっかくts使ってるので
型があるところは指定すると良いかなという提案です
```suggestion
    const handleClick = (e: MouseEvent) => {
```

Reply:
ありがとうございます。型をつけます！